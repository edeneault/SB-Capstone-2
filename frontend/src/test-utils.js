import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
