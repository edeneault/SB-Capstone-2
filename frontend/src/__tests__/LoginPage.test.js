import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { render } from "../test-utils";
import { initialState } from "../store-mock";
import { createMemoryHistory, createLocation } from "history";
import LoginPage from "../pages/LoginPage.js";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  const history = createMemoryHistory();
  const path = `/cart`;

  const match = {
    isExact: false,
    path,
    url: path,
  };
  const location = createLocation(match.url);

  test("renders LoginPage without crashing", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <LoginPage location={location} history={history} />
      </Provider>,
      { initialRoutes: [`/cart`] },
    );

    expect(getByText(/password/i)).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <LoginPage location={location} history={history} />
      </Provider>,
      { initialRoutes: [`/cart`] },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
