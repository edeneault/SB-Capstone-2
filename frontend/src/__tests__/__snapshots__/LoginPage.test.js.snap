// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="justify-content-md-center row"
    >
      <div
        class="col-md-6 col-12"
      >
        <h1>
          Sign In
        </h1>
        <form
          class=""
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="email"
            >
              Email Address
            </label>
            <input
              class="form-control"
              id="email"
              placeholder="Enter email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              id="password"
              placeholder="Enter password"
              type="password"
              value=""
            />
          </div>
          <button
            class="my-2 btn btn-primary"
            type="submit"
          >
            Sign In
          </button>
        </form>
        <div
          class="py-3 row"
        >
          <div
            class="col"
          >
            New Customer? 
            <a
              href="/register?redirect=/"
            >
              Register
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
